functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy

```diff
@@ -134,7 +134,7 @@ class ExecutionSpec extends BaseContainer {
             // Waits for all executions to get cleaned
             waitFor(ExecutionUtils.Retrievers.executionsForProject(client, projectName),
                 {it.isEmpty()},
-                WaitingTime.XTRA_EXCESSIVE)
+                WaitingTime.XTRA_EXCESSIVE, WaitingTime.MODERATE)
```

Comment:
It's not obvious to me how checking for something less frequently (`WaitingTime.MODERATE`) can fix this race condition in a *reliable manner*. IMO, a proper fix should handle this error condition and interpret it appropriately for the `waitFor` "acceptance" logic.

Reply:
Thanks for reviewing and I will double check it. What I could see is that this request make some queries to get executions data before responding it to the client side. On this test, this request is made each second and, in parallel, the server is performing an executions clean up in the project. So, for a one second period check, there is a big chance that some executions to be deleted after a request start and before it ends. I think there is no reason to have a check period for each one second, and increasing it to 5 seconds could be reasonable

Reply:
Changing the timing of the "check" can decrease the likelihood of the flake, but does not address the root cause.
For the reliable fix, you need to handle the error condition.

What error are you getting from the API? A 4xx or a 5xx?